src/main/java/com/aws/greengrass/lifecyclemanager/UnloadableService.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.lifecyclemanager;
+
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.lifecyclemanager.exceptions.ServiceLoadException;
+
+import java.util.Map;
+
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.VERSION_CONFIG_KEY;
+import static com.aws.greengrass.deployment.bootstrap.BootstrapSuccessCode.REQUEST_RESTART;
+
+public class UnloadableService extends GreengrassService {
+    ServiceLoadException serviceLoadException;
+
+    public UnloadableService(Topics topics, ServiceLoadException e) {
+        super(topics);
+        this.serviceLoadException = e;
+    }
+
+    @Override
+    protected void install() {
+        serviceErrored(serviceLoadException);
+    }
+
+    @Override
+    public boolean isBootstrapRequired(Map<String, Object> newServiceConfig) {
+        Topic versionTopic = getConfig().find(VERSION_CONFIG_KEY);
+        if (versionTopic == null) {
+            logger.atInfo().log("Bootstrap is required: current service version unknown");
+            return true;
+        }
+        if (!versionTopic.getOnce().equals(newServiceConfig.get(VERSION_CONFIG_KEY))) {
+            logger.atInfo().log("Bootstrap is required: service version changed");
+            return true;
+        }
+        if (!getConfig().toPOJO().equals(newServiceConfig)) {
```

Comment:
Will the service config topics always be equal to the newServiceConfig generated for deployment when customer hasn't changed anything about the unloadable service? Do we create any child config for the service at load time or runtime that might cause differences?